### PR TITLE
Static analysis wunused variable

### PR DESCRIPTION
**Pull request by sondredl**

### CODE REVIEW

File: editor/plugins/editor_plugin.h [hunk 1]
```diff
@@ -282,7 +282,6 @@ class EditorPlugins {
 public:
 	static int get_plugin_count() { return creation_func_count; }
 	static EditorPlugin *create(int p_idx) {
-		ERR_FAIL_INDEX_V(p_idx, creation_func_count, nullptr);
```
Comment (AThousandShips) [inline editor/plugins/editor_plugin.h:285]: These are valid and important checks, please restore

File: editor/plugins/editor_plugin.h [hunk 2]
```diff
@@ -292,7 +291,7 @@ class EditorPlugins {
 	}
 
 	static void add_create_func(EditorPluginCreateFunc p_func) {
-		ERR_FAIL_COND(creation_func_count >= MAX_CREATE_FUNCS);
+		(creation_func_count >= MAX_CREATE_FUNCS);
```
Comment (AThousandShips) [inline editor/plugins/editor_plugin.h:294]: This one too

File: editor/debugger/debug_adapter/debug_adapter_types.h [hunk 3]
```diff
@@ -67,7 +67,7 @@ struct Source {
 	String path;
 
 	void compute_checksums() {
-		ERR_FAIL_COND(path.is_empty());
+		(path.is_empty());
```
Comment (AThousandShips) [inline editor/debugger/debug_adapter/debug_adapter_types.h:70]: This is also invalid, please restore

**Review by AThousandShips — CHANGES_REQUESTED** [review]

**Comment by sondredl** [comment]
> Thank you for your contribution!
> 
> Please explain what this PR is intended to achieve, it makes a lot of changes that are against our code style norms and removes several actually important error checks that shouldn't be removed, we enforce using _both_ `virtual` and `override` for readability
> 
> These are very different changes and should (if they are valid in the first place) be made in separate PRs as they are completely unrelated
> 
> But this is something that should be discussed before making these changes, so please join our [developer chat](https://chat.godotengine.org/) and ask about these things

did nit realize i was making this pull request to the main repo. Im using the godot code base to experiment with static analysis tooling. Thanks for quick feedback! will close the pull request

**Comment by Calinou** [comment]
- Superseded by https://github.com/godotengine/godot/pull/98564.

**Pull request closed without merging**